docs/running-rules.md

```diff
@@ -9,7 +9,7 @@ import MoreHelp from "/src/components/MoreHelp"
 
 ## Introduction
 
-Semgrep reports findings in code based on rules. These rules use specific patterns to match code. When Semgrep matches a portion of code, it is reported as a finding. Findings may help you to catch issues of security, performance, correctness, and best practices. Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without need to write anything custom. The list below covers different kinds of Semgrep rules:
+Semgrep reports findings in code based on rules, which use specific patterns to match code. When Semgrep matches a portion of code, it is reported as a finding. Findings may help you to catch issues of security, performance, correctness, and best practices. Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without the need to write anything custom. The list below covers different kinds of Semgrep rules:
```

Comment:
Suggestion: 
Rules are used to detect specific patterns in a codebase. Semgrep reports all the matched locations as findings. Such findings cn help you catch issues of security...[the rest is unchanged]

Reply:
I understand your point @efortuna. I am still thinking that we are not paying our rules justice... they are recipes in the end, right? Recipes to find stuff in code (since it can be only one file and not necessarily the whole codebase, I would rather say "code", as it involves both single files and whole codebases in its meaning). What about:

Suggestion based on your suggestion:
Rules are instructions based on which Semgrep detects patterns in code. When Semgrep reports a code using mentioned rules, the detected code is called a finding. The process of scanning and detecting a piece of code is sometimes called *matching*, as Semgrep matched the code using rules to report a finding. Such findings can help you catch issues of security, performance, correctness, and best practices. Rules also have a form of patterns that you can define, or you can rely on rules created by the community or r2c. Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without the need to write anything custom. The list below covers different kinds of Semgrep rules:

Reply:
Original
"Rules also have a form of patterns that you can define, or you can rely on rules..."

Suggested
 -> You can define custom rules through Semgrep's rule syntax or you can rely on rules created by the community or r2c.

Reply:
Implemented!